app/forms/participants/participant_validation_form.rb

```diff
@@ -0,0 +1,143 @@
+        OpenStruct.new(id: "no", name: "No, I do not know my TRN"),
+        OpenStruct.new(id: "i_do_not_have", name: "I do not have a TRN"),
+      ]
+    end
+
+    def name_change_choices
+      [
+        OpenStruct.new(id: "yes", name: "Yes, I changed my name"),
+        OpenStruct.new(id: "no", name: "No, I have the same name"),
+      ]
+    end
+
+    def updated_record_choices
+      [
+        OpenStruct.new(id: "yes", name: "Yes, my name has been updated"),
+        OpenStruct.new(id: "no", name: "No, I need to update my name"),
+        OpenStruct.new(id: "i_do_not_know", name: "I’m not sure"),
+      ]
+    end
+
+    def name_not_updated_choices
+      [
+        OpenStruct.new(id: "register_previous_name", name: "Register for this programme using your previous name (you can update this later)"),
+        OpenStruct.new(id: "update_name", name: "Update your name with the Teaching Regulation Agency"),
+      ]
+    end
+
+    def pretty_date_of_birth
+      if date_of_birth.present?
+        date_of_birth.strftime("%d/%m/%Y")
+      end
+    end
+
+    def increment_validation_attempts
+      if validation_attempts.nil?
+        @validation_attempts = 1
+      else
+        @validation_attempts += 1
+      end
+    end
+
+  private
+
+    def trn_choice
+      if do_you_know_your_trn_choice.blank? || !do_you_know_your_trn_choice.in?(%w[yes no i_do_not_have])
+        errors.add(:do_you_know_your_trn_choice, :blank)
+      end
+    end
+
+    def name_change_choice
+      if have_you_changed_your_name_choice.blank? || !have_you_changed_your_name_choice.in?(%w[yes no])
+        errors.add(:have_you_changed_your_name_choice, :blank)
+      end
+    end
+
+    def confirm_updated_record_choice
+      if updated_record_choice.blank? || !updated_record_choice.in?(%w[yes no i_do_not_know])
+        errors.add(:updated_record_choice, :blank)
+      end
+    end
+
+    def confirm_name_not_updated_choice
+      if name_not_updated_choice.blank? || !name_not_updated_choice.in?(%w[register_previous_name update_name])
+        errors.add(:name_not_updated_choice, :blank)
+      end
+    end
+
+    def teacher_details
+      if trn.blank?
+        errors.add(:trn, :blank)
+      elsif trn.length < 5
+        errors.add(:trn, :too_short, count: 5)
+      elsif trn.length > 7
+        errors.add(:trn, :too_long, count: 7)
+      elsif trn !~ /\A\d+\z/
+        errors.add(:trn, :invalid)
+      end
+
+      if name.blank?
+        errors.add(:name, :blank)
+      end
+
+      if @date_of_birth_invalid
+        errors.add(:date_of_birth, :invalid)
+      elsif date_of_birth.blank?
+        errors.add(:date_of_birth, :blank)
+      elsif date_of_birth > Time.zone.now
+        errors.add(:date_of_birth, :in_the_future)
+      end
+
+      if national_insurance_number.present? && national_insurance_number !~ NINO_REGEX
+        errors.add(:national_insurance_number, :invalid)
+      end
+    end
```

Comment:
Should those be separate validations?

Reply:
The validations are grouped together to represent a page/step in the flow, not sure what you mean?

Reply:
Well, personally I prefer
```
validates :name, presence: true
```
to
```
if name.blank?
  errors.add(:name, :blank)
end
```
More importantly, in principle I like the idea of separating validations by fields, rather than by steps. It so happens that we don't need to (I think) run the validations for the same field in multiple steps, but as far as a general pattern goes, we totally could.

We can still group `validate` and `validates` statements by the steps they are run on, so in the top level view of the file you can see what's validated when quickly.

Reply:
Yes, of course, but this is a very specialised case. I also thought we might need to run more than one step group as we progressed through so this approach made a lot of sense to me.